src/main/java/jenkins/scm/api/SCMNavigator.java

```diff
@@ -67,10 +187,109 @@ public SCMNavigatorDescriptor getDescriptor() {
+     * persisted actions separately (for example {@link AbstractProject#getActions()} returns an immutable list,
+     * so there is no way to persist the actions from this method against those sub-classes, instead the actions
+     * need to be persisted by some side mechanism and then injected into the {@link Actionable#getAllActions()}
+     * through a {@link TransientActionFactory} ignoring the cognitive dissonance triggered by adding non-transient
+     * actions through a transient action factory... think of it instead as a {@code TemporalActionFactory} that adds
+     * actions that can change over time)
+     *
+     * @param owner    the owner of this {@link SCMNavigator}.
+     * @param event    the (optional) even to use when fetching the actions. Where the implementation is
+     *                 able to trust the event, it may use the event payload to reduce the number of
+     *                 network calls required to obtain the actions.
+     * @param listener the listener to report progress on.
+     * @return the map of {@link Action} instances to persist, keyed by the class of action. Keys with {@code null}
+     * values indicate actions that should be removed if present.
+     * @throws IOException          if an error occurs while performing the operation.
+     * @throws InterruptedException if any thread has interrupted the current thread.
+     * @since 2.0
+     */
+    @NonNull
+    public final List<Action> fetchActions(@NonNull SCMNavigatorOwner owner,
+                                           @CheckForNull SCMNavigatorEvent event,
+                                           @CheckForNull TaskListener listener)
+            throws IOException, InterruptedException {
+        return Util.fixNull(retrieveActions(owner, event, defaultListener(listener)));
+    }
+
+    /**
+     * SPI for {@link #fetchActions(SCMNavigatorOwner, SCMNavigatorEvent, TaskListener)}. Fetches any actions that
+     * should be persisted for objects related to the specified owner.
+     *
+     * @param owner    the owner of this {@link SCMNavigator}.
+     * @param event    the (optional) even to use when fetching the actions. Where the implementation is
+     *                 able to trust the event, it may use the event payload to reduce the number of
+     *                 network calls required to obtain the actions.
+     * @param listener the listener to report progress on.
+     * @return the map of {@link Action} instances to persist, keyed by the class of action. Keys with {@code null}
+     * values indicate actions that should be removed if present.
+     * @throws IOException          if an error occurs while performing the operation.
+     * @throws InterruptedException if any thread has interrupted the current thread.
+     * @since 2.0
+     */
+    @NonNull
+    public List<Action> retrieveActions(@NonNull SCMNavigatorOwner owner,
```

Comment:
I don't really see the difference between this and `fetchActions`? Maybe my speed reading isn't up to snuff?